src/main/java/org/opensearch/security/user/UserService.java

```diff
@@ -178,4 +202,79 @@ private String generatePassword() {
         String generatedPassword = "superSecurePassword";
         return generatedPassword;
     }
+
+    /**
+     * This function retrieves the auth token associated with a service account.
+     * Fails if the provided account is not a service account or account is not enabled.
+     *
+     * @param accountName A string representing the name of the account
+     * @return A string auth token
+     */
+    public String generateAuthToken(String accountName) throws IOException {
+
+        final SecurityDynamicConfiguration<?> internalUsersConfiguration = load(getUserConfigName(), false);
+
+        if (!internalUsersConfiguration.exists(accountName)) {
+            throw new UserServiceException(FAILED_ACCOUNT_RETRIEVAL_MESSAGE);
+        }
+
+        String authToken = null;
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            JsonNode accountDetails = mapper.readTree(internalUsersConfiguration.getCEntry(accountName).toString());
+            final ObjectNode contentAsNode = (ObjectNode) accountDetails;
+            SecurityJsonNode securityJsonNode = new SecurityJsonNode(contentAsNode);
+
+            if (securityJsonNode.get("isService").isNull()) { // If this is not a service account
+                throw new UserServiceException(AUTH_TOKEN_GENERATION_MESSAGE);
+            }
+            if (Objects.requireNonNull(securityJsonNode.get("isService").asString()).equalsIgnoreCase("false")) { // If this is not a service account
```

Comment:
Very Nitty; `requireNonNull` will throw an exception if the item is null, but the message might not be very useful without what is null if line number don't match up.

IMO I'd recommend using Optional with `.orElseThrow(...);` to prevent null pointers and let you clearly say what you wanted

Reply:
Oh, I did not realize this. I will swap over.